File: Java/app/src/main/java/dev/ogabek/java/MainActivity.java
package dev.ogabek.java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.KeyEvent;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView tempTextView, tempTextKey;
    private EditText textInputEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    private void initViews() {
        tempTextView = findViewById(R.id.tv_temp_text);
        tempTextKey = findViewById(R.id.tv_temp_key);
        textInputEditText = findViewById(R.id.et_text_input);

        textInputEditText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                tempTextView.setText(String.valueOf(s));
            }
        });

        textInputEditText.setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View v, int keyCode, KeyEvent event) {
                tempTextKey.setText(String.valueOf(keyCode));
                return false;
            }
        });

    }
}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/MainActivity.kt
package dev.ogabek.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText
import android.widget.TextView
import androidx.core.widget.addTextChangedListener
import android.view.KeyEvent

import android.view.View
import android.widget.Toast


class MainActivity : AppCompatActivity() {

    private lateinit var tempTextView: TextView
    private lateinit var tempTextKey: TextView
    private lateinit var textInputEditText: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView()

    }

    private fun initView() {
        tempTextView = findViewById(R.id.tv_temp_text)
        tempTextKey = findViewById(R.id.tv_temp_key)
        textInputEditText = findViewById(R.id.et_text_input)

        textInputEditText.addTextChangedListener {
            tempTextView.text = it.toString()
        }

        textInputEditText.setOnKeyListener { _, keyCode, _ ->

            tempTextKey.text = keyCode.toString()

            if (keyCode == 66) {
                Toast.makeText(applicationContext, "You Pressed Enter Key", Toast.LENGTH_SHORT).show()
                textInputEditText.setText("")
            }

            false
        }

    }
}